jenkins/opensearch-dashboards/distribution-build.jenkinsfile

```diff
@@ -863,6 +863,7 @@ pipeline {
         }
         success {
             node(AGENT_X64) {
+                checkout scm
```

Comment:
Why do we need this? It does that by default right?

Reply:
Since to this part we are not bringing the stashed folder (jenkins workspace) we run `checkout scm` to use the `inputManifestPath: "manifests/$INPUT_MANIFEST"`.

Reply:
Without that too it is available right? I believe jenkins autoclones the repo by default.

Reply:
Sayali, we need it, we even have this for `failure `  case https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/distribution-build.jenkinsfile#L953
Error without `checkout scm`:
```
Also:   org.jenkinsci.plugins.workflow.actions.ErrorAction$ErrorId: 875e63c3-7f75-482b-817a-894cbd17e88d
java.io.FileNotFoundException: /var/jenkins/workspace/pgodithi/distribution-build-opensearch/manifests/2.8.0/opensearch-2.8.0.yml does not exist.
	at org.jenkinsci.plugins.pipeline.utility.steps.conf.ReadYamlStep$Execution.doRun(ReadYamlStep.java:185)
	at org.jenkinsci.plugins.pipeline.utility.steps.AbstractFileOrTextStepExecution.run(AbstractFileOrTextStepExecution.java:29)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Finished: FAILURE
```

Reply:
Sorry figured out the issue now. Since this is in post stage the environment is completely different. My bad. 
Thanks!